src/screens/Wallet.js

```diff
@@ -195,16 +254,23 @@ class Wallet extends React.Component {
     this.props.history.push('/addresses/');
   }
 
+  // Trigger a render when we sign a token
+  updateTokenSignature = (value) => {
+    this.setState({
+      hasTokenSignature: value,
+    });
+  }
+
   retryDownload = (e, tokenId) => {
     e.preventDefault();
     const balanceStatus = get(
-      this.props.tokensBalance,
-      `${tokenId}.status`,
+      this.props.tokenBalance,
```

Comment:
Why did you change this? It looks like it's `tokensBalance`

```
const mapStateToProps = (state) => {
  return {
    selectedToken: state.selectedToken,
    entireState: state,
    tokensHistory: state.tokensHistory,
    tokensBalance: state.tokensBalance,
    tokens: state.tokens,
    wallet: state.wallet,
    useWalletService: state.useWalletService,
  };
};
```

Reply:
This is indeed a bug, fixed in 2f91a8ad674689fffb0ee4645163f8012c438a2e